dart/lib/src/client.dart

```diff
@@ -188,15 +109,63 @@ abstract class SentryClient {
     options.httpClient?.close();
   }
 
-  @override
-  String toString() => '$SentryClient("$postUri")';
+  SentryEvent _processEvent(
+    SentryEvent event, {
+    dynamic hint,
+    List<EventProcessor> eventProcessors,
+  }) {
+    for (final processor in eventProcessors) {
+      event = processor(event, hint);
+    }
+    return event;
+  }
 
-  @protected
-  List<int> bodyEncoder(Map<String, dynamic> data, Map<String, String> headers);
+  SentryEvent _applyScope(
+      {@required SentryEvent event, @required Scope scope}) {
+    if (scope != null) {
+      // Merge the scope transaction.
+      if (event.transaction == null) {
+        event = event.copyWith(transaction: scope.transaction);
+      }
+
+      // Merge the user context.
+      if (event.userContext == null) {
+        event = event.copyWith(userContext: scope.user);
+      }
+
+      // Merge the scope fingerprint.
+      if (event.fingerprint == null) {
+        event = event.copyWith(fingerprint: scope.fingerprint);
+      }
+
+      // Merge the scope breadcrumbs.
+      if (event.breadcrumbs == null) {
+        event = event.copyWith(breadcrumbs: scope.breadcrumbs);
+      }
+
+      // Merge the scope tags.
+      if (event.tags == null) {
+        event = event.copyWith(tags: scope.tags);
+      }
+
+      // Merge the scope extra.
+      if (event.extra == null) {
+        event = event.copyWith(extra: scope.extra);
+      }
+
+      // Merge the scope level.
+      if (event.level == null) {
+        event = event.copyWith(level: scope.level);
+      }
+    }
+    return event;
+  }
+
+  @override
+  String toString() => '$SentryClient("${options.dsn}")';
```

Comment:
mm not against it, but I don't see how this is useful :)

Reply:
I don't know :) until know the postUri was returned... what kind of infos should we return here ? Do we need it ?

Reply:
I dont think we need to overwrite the toString for this tbh, makes more sense for data classes only IMO

Reply:
✅